nucypher/characters/lawful.py

```diff
@@ -1237,10 +1234,7 @@ def __prune_datastore(self) -> None:
                 self.log.debug(f"Pruned {result} policy arrangements.")
 
         try:
-            with self.datastore.query_by(DatastoreTreasureMap,
-                                         filter_field='expiration',
-                                         filter_func=lambda expiration: expiration <= now,
-                                         writeable=True) as expired_treasure_maps:
+            with find_expired_treasure_maps(self.datastore, now) as expired_treasure_maps:
```

Comment:
Wow, very nice.

Reply:
Looks very clean, and I much prefer this pattern in this file.

Reply:
+1 - that's great!